app/src/main/java/org/p2p/wallet/striga/user/StrigaStorage.kt

```diff
@@ -1,61 +1,80 @@
 ) : StrigaStorageContract {
 
-    override var userStatus: StrigaUserStatusDetails?
-        get() = sharedPreferences.getString(KEY_USER_STATUS, null)?.let(gson::fromJsonReified)
-        set(value) {
-            sharedPreferences.edit {
-                if (value == null) {
-                    remove(KEY_USER_STATUS)
-                } else {
-                    putString(KEY_USER_STATUS, gson.toJson(value))
-                }
-            }
-        }
-
-    override var smsExceededVerificationAttemptsMillis: MillisSinceEpoch
-        get() = sharedPreferences.getLong(KEY_SMS_EXCEEDED_VERIFICATION_ATTEMPTS_MILLIS, 0L)
-        set(value) = sharedPreferences.edit(true) {
-            putLong(KEY_SMS_EXCEEDED_VERIFICATION_ATTEMPTS_MILLIS, value)
-        }
-
-    override var smsExceededResendAttemptsMillis: MillisSinceEpoch
-        get() = sharedPreferences.getLong(KEY_SMS_EXCEEDED_RESEND_ATTEMPTS_MILLIS, 0L)
-        set(value) = sharedPreferences.edit(true) {
-            putLong(KEY_SMS_EXCEEDED_RESEND_ATTEMPTS_MILLIS, value)
-        }
+    override var userStatus: StrigaUserStatusDetails? by ObjectEncryptedPreference(
+        preferences = encryptedPrefs,
+        keyProvider = { KEY_USER_STATUS },
+        type = StrigaUserStatusDetails::class,
+        nullIfMappingFailed = true
+    )
+
+    override var userWallet: StrigaUserWallet? by ObjectEncryptedPreference(
+        preferences = encryptedPrefs,
+        keyProvider = { KEY_USER_WALLET },
+        type = StrigaUserWallet::class,
+        nullIfMappingFailed = true
+    )
+
+    override var fiatAccount: StrigaFiatAccountDetails? by ObjectEncryptedPreference(
+        preferences = encryptedPrefs,
+        keyProvider = { KEY_FIAT_ACCOUNT_DETAILS },
+        type = StrigaFiatAccountDetails::class,
+        nullIfMappingFailed = true
+    )
+
+    override var smsExceededVerificationAttemptsMillis: MillisSinceEpoch by LongPreference(
+        preferences = encryptedPrefs,
+        keyProvider = { KEY_SMS_EXCEEDED_VERIFICATION_ATTEMPTS_MILLIS },
+        defaultValue = 0
+    )
+
+    override var smsExceededResendAttemptsMillis: MillisSinceEpoch by LongPreference(
+        preferences = encryptedPrefs,
+        keyProvider = { KEY_SMS_EXCEEDED_RESEND_ATTEMPTS_MILLIS },
+        defaultValue = 0
+    )
 
     override fun hideBanner(banner: StrigaKycStatusBanner) {
-        sharedPreferences.edit(true) {
-            putBoolean(KEY_USER_BANNER_IS_HIDDEN_PREFIX + banner.name, true)
-        }
+        val hiddenBanners = encryptedPrefs.getStringSet(KEY_USER_BANNER_IS_HIDDEN)
+        encryptedPrefs.putStringSet(KEY_USER_BANNER_IS_HIDDEN, hiddenBanners + banner.bannerId.toString())
     }
 
     override fun isBannerHidden(banner: StrigaKycStatusBanner): Boolean {
-        return sharedPreferences.getBoolean(KEY_USER_BANNER_IS_HIDDEN_PREFIX + banner.name, false)
+        return banner.toString() in encryptedPrefs.getStringSet(KEY_USER_BANNER_IS_HIDDEN)
```

Comment:
can you separate this in two lines ? to have ability to take it easy on debugging ?

Reply:
yep, will do